src/safeds/ml/classical/regression/_arima.py

```diff
@@ -0,0 +1,209 @@
+
+
+class ArimaModelRegressor:
+    """Auto Regressive Integrated Moving Average Model."""
+
+    def __hash__(self) -> int:
+        """
+        Return a deterministic hash value for a regressor.
+
+        Returns
+        -------
+        hash : int
+            The hash value.
+        """
+        return xxhash.xxh3_64(
+            self.__class__.__qualname__.encode("utf-8")
+            + (1 if self.is_fitted() else 0).to_bytes(1)
+            + (bytes((9, 9, 9)) if self._order is None else bytes(self._order)),
+        ).intdigest()
+
+    def __init__(self) -> None:
+        # Internal state
+        self._arima: ARIMA | None = None
+        self._order: tuple[int, int, int] | None = None
+        self._fitted = False
+
+    def fit(self, time_series: TimeSeries) -> ArimaModelRegressor:
+        """
+        Create a copy of this ARIMA Model and fit it with the given training data.
+
+        This ARIMA Model is not modified.
+
+        Parameters
+        ----------
+        time_series:
+            The time series containing the target column, which will be used.
+
+        Returns
+        -------
+        fitted_arima:
+            The fitted ARIMA Model.
+
+        Raises
+        ------
+        LearningError
+            If the training data contains invalid values or if the training failed.
+        NonTimeSeriesError
+            If the table is not a TimeSeries object.
+        NonNumericColumnError
+            If the training data contains non-numerical values.
+        MissingValuesColumnError
+            If the training data contains missing values.
+        DatasetMissesDataError
+            If the training data contains no rows.
+        """
+        if not isinstance(time_series, TimeSeries) and isinstance(time_series, Table):
+            raise NonTimeSeriesError
+        if time_series.number_of_rows == 0:
+            raise DatasetMissesDataError
+        if not time_series.target.type.is_numeric():
+            raise NonNumericColumnError(time_series.target.name)
+        if time_series.target.has_missing_values():
+            raise MissingValuesColumnError(
+                time_series.target.name,
+                "You can use the Imputer to replace the missing values based on different strategies.\nIf you want to"
+                "remove the missing values entirely you can use the method "
+                "`TimeSeries.remove_rows_with_missing_values`.",
+            )
+        fitted_arima = ArimaModelRegressor()
+        p = d = q = range(2)
+        pdq = list(itertools.product(p, d, q))
+        best_aic = float("inf")
+        best_model = None
+        # best param will get overwritten
+        best_param = (0, 0, 0)
+        for param in pdq:
+            # Create and fit an ARIMA model with the current parameters
+            mod = ARIMA(time_series.target._data.values, order=param)
+
+            # I wasnt able to invoke an learning Error
+            # Add try catch when an learning error is found
```

Comment:
Is this a todo comment?

Reply:
no, its a reminder for me, where to change something, when some1 finds a way